### PR TITLE
cassandra@3 3.11 (new formula)

### PR DESCRIPTION
**Pull request by ghost**
Adding a versioned Formulae for Cassandra@3.11 (which disappeared after
the recent upgrade to 4.0.0).

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by carlocab** [comment]
Thanks for the PR. Does this meet our requirements for versioned formulae at https://docs.brew.sh/Versions?

**Comment by alecl** [comment]
Is something amiss with the 11-arm64 CI environment? This bottle is missing there which is a dependency:

Error: openjdk@8: no bottle available!

**Comment by SMillerDev** [comment]
OpenJDK 8 does not exist for ARM.

File: Formula/cassandra@3.11.rb
Comment (iMichka) [inline Formula/cassandra@3.11.rb:17]: Please add:
`depends_on arch: :x86_64 # openjdk@8 doesn't support ARM`

**Comment by c3-camille-masset** [comment]
@iMichka - I added the line you suggested, thanks for the tip. However the CI (for version `11-arm64`) is still failing with the following error:
```
brew audit cassandra@3.11 --online --new-formula
Warning: The x86_64 architecture is required for this software.
FAILED
Installing 'bundler' gem
cassandra@3.11:
  * New formulae in homebrew/core should not have a `bottle do` block
```
How to make it consider that the formulae is not new? Thanks for the help!

**Comment by carlocab** [comment]
It's new because `cassandra@3.11.rb` is a file that didn't exist before. Fix it by removing the `bottle` block.

**Comment by SMillerDev** [comment]
I think this should be `cassandra@3` though, there are no other 3.x releases supported.

**Comment by c3-camille-masset** [comment]
> I think this should be `cassandra@3` though, there are no other 3.x releases supported.

I've renamed the file and changed the name of the Ruby class.

**Comment by SMillerDev** [comment]
If you're here, can you still answer https://github.com/Homebrew/homebrew-core/pull/82177#issuecomment-889882063

**Comment by c3-camille-masset** [comment]
> Thanks for the PR. Does this meet our requirements for versioned formulae at https://docs.brew.sh/Versions?

I think it does (this is my first contribution to Homebrew though, I hope I have not missed anything going through the bullet list in the page you linked).

**Comment by SMillerDev** [comment]
Okay, do you have a link to the EOL date, or the continued development for Cassandra 3? Would be useful to have here for future reference.

**Comment by c3-camille-masset** [comment]
> Okay, do you have a link to the EOL date, or the continued development for Cassandra 3? Would be useful to have here for future reference.

I could not find it on Cassandra website, but according to Instaclustr, the EOL is April 30, 2022 (critical patches until April 30, 2023): https://www.instaclustr.com/instaclustrs-support-for-cassandra/

**Comment by c3-camille-masset** [comment]
@iMichka - can you please give it another review? 😊

**Comment by SMillerDev** [comment]
Can you ask the Cassandra developers about their EOL policy and add a `deprecated!` or `disabled!` call into the formula with the depreciation or EOL date?

**Comment by Aqualon** [comment]
I tried to use the formula and cassandra doesn't start, because `CASSANDRA_CONF` points to `/usr/local/etc/cassandra/`, but I only had `/usr/local/etc/cassandra@3/` after `brew install --build-from-source cassandra@3`.

And I got again the jdk8 crash I once reported in https://github.com/Homebrew/homebrew-core/issues/66462

```
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x0000000115f45f0e, pid=70830, tid=0x0000000000006a03
#
# JRE version: OpenJDK Runtime Environment (8.0_302) (build 1.8.0_302-bre_2021_07_21_23_11-b00)
# Java VM: OpenJDK 64-Bit Server VM (25.302-b00 mixed mode bsd-amd64 compressed oops)
# Problematic frame:
# V  [libjvm.dylib+0x545f0e]  Unsafe_GetByte+0x5c
#
# Failed to write core dump. Core dumps have been disabled. To enable core dumping, try "ulimit -c unlimited" before starting Java again
#
# An error report file with more information is saved as:
# /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/hs_err_pid70830.log
#
# If you would like to submit a bug report, please visit:
#   http://bugreport.java.com/bugreport/crash.jsp
#
Abort trap: 6
```

What worked for me in the end was to use 
```JAVA_HOME=/Library/Java/JavaVirtualMachines/adoptopenjdk-8.jdk/Contents/Home CASSANDRA_CONF=/usr/local/etc/cassandra@3/ /usr/local/opt/cassandra@3/bin/cassandra -f``` 
to start it

**Comment by mzuehlke** [comment]
The EOL dates for the different versions should be updated as agree in the `[DISCUSS] Releases after 4.0` email thread.

* 4.0: Fully supported until April 2023 and high severity bugs until April 2024 (2 year full, 1 year bugfix)
* 3.11: Fully supported until April 2022 and high severity bugs until April 2023 (1 year full, 1 year bugfix).
* 3.0: Supported for high severity correctness/performance bugs until April 2022 (1 year bugfix)
* 2.2+2.1: EOL immediately.

https://issues.apache.org/jira/browse/CASSANDRA-16556

**Comment by SMillerDev** [comment]
I am not allowed to change this PR so that's up to @c3-camille-masset to add

**Comment by dzlab** [comment]
Any updates on this PR when it can be merge?

**Comment by SMillerDev** [comment]
As soon as the last change is done it can be merged IYAM.

**Comment by c3-camille-masset** [comment]
> The EOL dates for the different versions as agree in the `[DISCUSS] Releases after 4.0` email thread.
> 
> * 4.0: Fully supported until April 2023 and high severity bugs until April 2024 (2 year full, 1 year bugfix)
> * 3.11: Fully supported until April 2022 and high severity bugs until April 2023 (1 year full, 1 year bugfix).
> * 3.0: Supported for high severity correctness/performance bugs until April 2022 (1 year bugfix)
> * 2.2+2.1: EOL immediately.
> 
> https://issues.apache.org/jira/browse/CASSANDRA-16556

Thanks @mzuehlke! 

@SMillerDev - I put a `deprecate!` for version 3.11 in the formula.

**Review by SMillerDev — APPROVED** [review]

File: Formula/cassandra@3.rb
Comment (carlocab) [inline Formula/cassandra@3.rb:13]: I'm a bit surprised this is used, since `cython` uses Python 3.9.

Comment (c3-camille-masset) [reply Formula/cassandra@3.rb:13]: I took the formula as it was before Cassandra 4.0 was made available in Homebrew.

Comment (carlocab) [reply Formula/cassandra@3.rb:13]: I'm aware. I would ask the same question about the last version of the formula.

**Review by iMichka — APPROVED** [review]

**Pull request closed without merging**
